presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -73,4 +121,81 @@ public void testDropFunctionInvalidFunctionName()
+        @Language("SQL") String createFunctionDouble = "CREATE FUNCTION testing.common.array_append(a array<double>, x double)\n" +
+                "RETURNS array<double>\n" +
+                "RETURN concat(a, array[x])";
+        @Language("SQL") String createFunctionRand = "CREATE FUNCTION testing.common.rand()\n" +
+                "RETURNS double\n" +
+                "RETURN rand()";
+        String createFunctionIntFormatted = "CREATE FUNCTION testing.common.array_append (\n" +
+                "   \"a\" ARRAY(integer),\n" +
+                "   \"x\" integer\n" +
+                ")\n" +
+                "RETURNS ARRAY(integer)\n" +
+                "COMMENT ''\n" +
+                "LANGUAGE SQL\n" +
+                "NOT DETERMINISTIC\n" +
+                "CALLED ON NULL INPUT\n" +
+                "RETURN \"concat\"(a, ARRAY[x])";
+        String createFunctionDoubleFormatted = "CREATE FUNCTION testing.common.array_append (\n" +
+                "   \"a\" ARRAY(double),\n" +
+                "   \"x\" double\n" +
+                ")\n" +
+                "RETURNS ARRAY(double)\n" +
+                "COMMENT ''\n" +
+                "LANGUAGE SQL\n" +
+                "NOT DETERMINISTIC\n" +
+                "CALLED ON NULL INPUT\n" +
+                "RETURN \"concat\"(a, ARRAY[x])";
+        String createFunctionRandFormatted = "CREATE FUNCTION testing.common.rand ()\n" +
+                "RETURNS double\n" +
+                "COMMENT ''\n" +
+                "LANGUAGE SQL\n" +
+                "NOT DETERMINISTIC\n" +
+                "CALLED ON NULL INPUT\n" +
+                "RETURN \"rand\"()";
+        String parameterTypeInt = "ARRAY(integer), integer";
+        String parameterTypeDouble = "ARRAY(double), double";
+
+        assertQuerySucceeds(createFunctionInt);
+        assertQuerySucceeds(createFunctionDouble);
+        assertQuerySucceeds(createFunctionRand);
+
+        MaterializedResult rows = computeActual("SHOW CREATE FUNCTION testing.common.array_append");
+        assertEquals(rows.getRowCount(), 2);
+        assertEquals(rows.getMaterializedRows().get(0).getFields(), ImmutableList.of(createFunctionDoubleFormatted, parameterTypeDouble));
+        assertEquals(rows.getMaterializedRows().get(1).getFields(), ImmutableList.of(createFunctionIntFormatted, parameterTypeInt));
+
+        rows = computeActual("SHOW CREATE FUNCTION testing.common.array_append(array(int), int)");
+        assertEquals(rows.getRowCount(), 1);
+        assertEquals(rows.getMaterializedRows().get(0).getFields(), ImmutableList.of(createFunctionIntFormatted, parameterTypeInt));
+
+        rows = computeActual("SHOW CREATE FUNCTION testing.common.rand()");
+        assertEquals(rows.getMaterializedRows().get(0).getFields(), ImmutableList.of(createFunctionRandFormatted, ""));
+
+        assertQueryFails("SHOW CREATE FUNCTION testing.common.array_append()", "Function not found: testing\\.common\\.array_append\\(\\)");
```

Comment:
Also add test for things like 'ARRAY_AGG'. What happens when someone tries that?

Reply:
test added